core/Command/Preview/Repair.php

```diff
@@ -176,107 +151,142 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 		$progressBar->maxSecondsBetweenRedraws(0.2);
 		$progressBar->start();
 
-		foreach ($directoryListing as $oldPreviewFolder) {
-			pcntl_signal_dispatch();
-			$name = $oldPreviewFolder->getName();
-			$time = (new \DateTime())->format('H:i:s');
-			$section1->writeln("$time Migrating previews of file with fileId $name …");
-			$progressBar->display();
+		$limit = 1000;
 
-			if ($this->stopSignalReceived) {
-				$section1->writeln("$time Stopping migration …");
-				return 0;
-			}
-			if (!$oldPreviewFolder instanceof Folder) {
-				$section1->writeln("         Skipping non-folder $name …");
-				$progressBar->advance();
-				continue;
-			}
-			if ($name === 'old-multibucket') {
-				$section1->writeln("         Skipping fallback mount point $name …");
-				$progressBar->advance();
-				continue;
-			}
-			if (in_array($name, ['a', 'b', 'c', 'd', 'e', 'f'])) {
-				$section1->writeln("         Skipping hex-digit folder $name …");
-				$progressBar->advance();
-				continue;
-			}
-			if (!preg_match('!^\d+$!', $name)) {
-				$section1->writeln("         Skipping non-numeric folder $name …");
-				$progressBar->advance();
-				continue;
-			}
+		$hasEntries = true;
+		do {
+			$qb = $this->db->getQueryBuilder();
+			$cursor = $qb->select('name')->from('filecache')
+				->where($qb->expr()->eq('parent', $qb->createNamedParameter($currentPreviewFolder->getId())))
```

Comment:
Wouldn't it make sense to remember the highest fileID from the previous loop run and put it in here as well?

```suggestion
				->where($qb->expr()->eq('parent', $qb->createNamedParameter($currentPreviewFolder->getId())))
				->andWhere($qb->expr()->gt('fileid', $qb->createNamedParameter($previousHighestFileId)))
```